### PR TITLE
Sumeragi: peers in set b related TODO fixes

### PR DESCRIPTION
**Pull request by eadventurous**

### CODE REVIEW

File: iroha/src/sumeragi.rs [hunk 1]
```diff
@@ -103,6 +103,9 @@ impl Sumeragi {
                 for peer in self.network_topology.validating_peers() {
                     send_futures.push(message.clone().send_to(peer));
                 }
+                for peer in self.network_topology.peers_set_b() {
+                    send_futures.push(message.clone().send_to(peer));
+                }
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:108]: Maybe merge these two sets first and iterate over the whole iter after?

File: iroha/src/sumeragi.rs [hunk 2]
```diff
@@ -381,45 +384,47 @@ impl Sumeragi {
                     .expect("Failed to get System Time.")
             && tx_receipt.is_valid(&self.network_topology)
         {
-            self.transactions_awaiting_receipts
+            let transaction_was_waiting = self
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:387]: unnecessary local variable, if you made it to make condition more verbose and clear, it's better to encapsulate it somewhere.

File: iroha/src/sumeragi.rs [hunk 3]
```diff
@@ -529,24 +537,27 @@ impl Sumeragi {
...
-        if valid_signatures.len() >= self.network_topology.min_votes_for_commit()
-            && proxy_tail_signatures.len() == 1
-        {
-            block.signatures.clear();
-            block.signatures.append(&valid_signatures);
-            let block = block.validate(&*self.world_state_view.read().await)?;
-            let hash = block.hash();
-            self.blocks_sender.write().await.send(block).await;
-            self.next_round(hash).await;
+        let voting_block = self.voting_block.write().await.clone();
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:540]: unnecessary local variable

Comment (eadventurous) [reply iroha/src/sumeragi.rs:540]: I wanted to to skip it at first but it turns out to be necessary.
` if let Some(voting_block) = self.voting_block.write().await.clone() {
    |                                     -------------------------------
    |                                     |
    |                                     immutable borrow occurs here
    |                                     a temporary with access to the immutable borrow is created here ...
...
550 |                 self.next_round(hash).await;
    |                 ^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
`

**Pull request merged**